app/models/payment_providers/base_provider.rb

```diff
@@ -26,6 +26,14 @@ class BaseProvider < ApplicationRecord
     validates :name, presence: true
 
     settings_accessors :webhook_secret, :success_redirect_url
+
+    def determine_payment_status(payment_status)
+      return :pending if self.class::PENDING_STATUSES.include?(payment_status)
+      return :succeeded if self.class::SUCCESS_STATUSES.include?(payment_status)
+      return :failed if self.class::FAILED_STATUSES.include?(payment_status)
+
+      payment_status
```

Comment:
nil or payment_status. If not nil, this will break when creating the payment, but it should break if we don't know the status?